language-design.md

```diff
@@ -26,32 +30,32 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
----
+This is a bit of a tough question because what makes a language well or poorly designed sometimes depends on the metrics that we use. One can argue R is a well-designed language, but Tim Smith found it difficult to learn, and so disliked it even though he now regrets the tone he took in intitial criticisms. [Smith, 2016] Perl is very difficult to learn as well, and yet is still relevant today as a programming language. [Stefik _et al._, 2011] Therefore, metrics like ease-of-use for a novice programmer can be subjective and are one symptom of a language. That being said, I think there are some things I use when I try to figure out whether a language is well-designed or not. I definitely will judge it based on whether the rules feel consistent. Python, C++, and R all have different means and rules for assignments, but once you know the rules you can always assign something. Similarly, if it's consistent than I also feel like it's a little predictable - little syntax things I might have to look up, but I can follow what's happening in a program. A poorly designed language, I cannot.
+
 
 **Question**
 
-How might the themes of _Growing a Language_ relate to notions of fluency from last week's
-class?
+How might the themes of _Growing a Language_ relate to ideas from the Fowler reading?
 
 **Response**
 
----
+More obviously, the idea of limited expressiveness that Fowler discusses is very related to _Growing a Language_ and the ways that Steele uses limited expressiveness to make his points. Of using only one syllable words, unless defined first, he says "in truth, the words of one syllable form quite a rich vocabulary, with which you can say many things." [Steele, 1995] He is able to use a small language and express many things quite fluently, or at least the bare minimum things that he needs - note that Steele doesn't define anything he doesn't use. Similarly, Fowler characterizes DSls with limited expressiveness, and that it doesn't have extraneous features, just what it needs to support its domain. [Fowler, 2010] There are a few other design principles Fowler and Steele both highlight. First, the importance of user input. In Fowler, he talks about the benefits of DSLs as (and needing) communication with domain experts - they are the ones with the specific knowledge about how to make the DSL most helpful. [Fowler, 2010] This is echoed in Steele's discussion of building things as a bazaar, and gathering lots of helping hands and inputs. [Steele, 1995] That also builds into another point Fowler makes about improving development productivity. Both readings thematically are trying to improve an experience and the tools one uses, in order to also improve what someone can do with a programming language.
```

Comment:
The beginning of the talk where Steele has to define everything was rough, but I suppose that's representing the growing pains of developing a language from scratch. I do agree that after all that was out of the way, his small vocabulary did not limit him whatsoever in trying to get his point across; in fact, it was refreshingly short with little embellishments.  I agree that this is something that we should aspire to do with our DSLs.